File: apigateway_getting_started/create_api_client_resource.py
import botocore
import boto3
import logging

logger = logging.getLogger()
apig = boto3.client("apigateway")

try:
    result = apig.create_resource(
        restApiId="abcd1234", parentId="efg567", pathPart="pets"
    )
except botocore.exceptions.ClientError as error:
    logger.exception("The '/pets' resource setup failed: %s.", error)
    raise
attribute = [
    "id",
    "parentId",
    "pathPart",
    "path",
]
filtered_result = {key: result[key] for key in attribute}
print(filtered_result)

File: apigateway_getting_started/create_api_client_resource.js
import { APIGatewayClient, CreateResourceCommand } from "@aws-sdk/client-api-gateway";
(async function () {
    const apig = new APIGatewayClient({ region: "us-east-1" });
    const command = new CreateResourceCommand({
        restApiId: 'abcd1234',
        parentId: 'efg567',
        pathPart: 'pets'
    });
    try {
        const results = await apig.send(command)
        console.log(results)
    } catch (err) {
        console.log("The '/pets' resource setup failed:\n", err)
    }
})();